spotify/data_source_album.go

```diff
@@ -39,6 +39,18 @@ func dataSourceAlbum() *schema.Resource {
 				Elem:        &schema.Schema{Type: schema.TypeString},
 				Description: "The spotify IDs of the artists",
 			},
+			"tracks": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "The spotify IDs of the tracks",
+			},
+			"track_names": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "The names of the tracks",
+			},
```

Comment:
Maybe instead of having 2 lists of strings, instead we could have a complex list

```suggestion
			"tracks": {
				Type:        schema.TypeList,
				Computed:    true,
				Elem:        &schema.Resource{
					Schema: map[string]*schema.Schema{
						"name": {
							Type:        schema.TypeString,
							Description: "The name of the track",
						},
						"id": {
							Type:        schema.TypeString,
							Description: "The name of the track",
						},
					},
					Description: "The spotify IDs of the tracks",
				},
			}
```

Reply:
You can still get the individual items using splat expressions, so instead of `data.spotify_album.foo.track_names` it would become `data.spotify_album.foo.tracks[*].name`